[Term Entry] Swift Protocols Equatable (#2999)
Codecademy/docs: content/swift/concepts/protocols/terms/equatable/equatable.md

---
Title: 'Equatable'
Description: 'Enables two instances of a type to be compared for equality.'
Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:
  - 'Protocols'
  - 'Equality'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

**`Equatable`** is a [protocol](https://www.codecademy.com/resources/docs/swift/protocols) that enables two instances of a type to be compared for equality using the `==` or `!=` operators.

## Syntax

In a straightforward case, for a type to conform to `Equatable`, all of its properties must also conform to `Equatable`:

```pseudo
struct myStruct: Equatable {
  // properties that conform to Equatable
}
```

> In this case, when checking for equality between two instances of the `myStruct` [structure](https://www.codecademy.com/resources/docs/swift/structures), all properties will be compared.

Reviewer: ```suggestion
When checking for equality between two instances of the `myStruct` [structure](https://www.codecademy.com/resources/docs/swift/structures), all properties will be compared.
```

### Syntax using a custom `==` method

If a customised approach is needed for equality comparison, then a custom `==` method can be included so that the type can conform to `Equatable`:

```pseudo
struct myStruct: Equatable {
  // properties

  static func == (lhs: myStruct, rhs: myStruct) -> Bool {
    // custom logic for equality comparison
  }
}
```

A custom `==` method can be helpful if only specific properties need to be compared, or in cases when some of the properties don't conform to `Equatable`.

## Example

In the example below, a `TeaCup` structure is declared with three properties `color`, `volume`, and `unit`. These properties all conform to the `Equatable` protocol so instances of `TeaCup` can be compared using the `==` operator.

Below the `TeaCup` structure, two instances of `TeaCup` are created and compared for equality:

> ```swift

Reviewer 2: looks like this code is using tabs, indentations should be done with 2 spaces. Please update both examples

Author: @caupolicandiaz Thanks so much for the review! I updated the code examples to have 2 spaces and also committed the other suggestions.

struct TeaCup: Equatable {
    var color: String
    var volume: Int
    var unit: Unit

    enum Unit {
        case cups, ml
    }
}

let blueCup = TeaCup(color: "Blue", volume: 237, unit: .ml)
let pinkCup = TeaCup(color: "Pink", volume: 1, unit: .cups)

if blueCup == pinkCup {
  print("Teacups are equal.")
}
else {
  print("Teacups are not equal.")
}
```

> Because the `blueCup` and `pinkCup` properties are not equal, this will output the following:

Reviewer: ```suggestion
Because the `blueCup` and `pinkCup` properties are not equal, the following output occurs:
```
